### PR TITLE
Fix/prediction

### PR DESCRIPTION
**Pull request by henriettemiko**
- added more to .gitignore
- generalized naming of processed file
- changed computation of consensus
- renamed loop index

question: should final prediction be labels[consensus] (most frequent class prediction over spectrograms) or labels[idx] (maximum of mean probabilities over spectrograms for each class)?

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -3,8 +3,9 @@
 __pycache__/params.cpython-36.pyc
 __pycache__/VGGish_model.cpython-36.pyc
 __pycache__/vggish_params.cpython-36.pyc
-.DS_Store
```
Comment (birdNicolas) [inline .gitignore:6]: warum raus? Ist nen typisches Temp-File für den Mac.

Comment (lucasmllr) [reply .gitignore:6]: glaube einfach damit es nicht im repo liegt. lässt man das sonst drin?

Comment (birdNicolas) [reply .gitignore:6]: ich meinte in dem Pull-Request, wird `.DS_Store` aus der `.gitignore` rausgenommen, dh. `.DS_Store` würde danach im Repo als potentieller add-Kandidat gelistet werden.

Comment (henriettemiko) [reply .gitignore:6]: .DS_Store war doppelt drin in der .gitignore, ist also immer noch drin

Comment (birdNicolas) [reply .gitignore:6]: ahh. alles klar :)

File: predict.py [hunk 2]
```diff
@@ -78,21 +78,27 @@ def predict(net, labels, files, params):
...
         pre.preprocess(file, processed)
         data = vggish_input.wavfile_to_examples(processed)
         data = torch.from_numpy(data).unsqueeze(1).float()
         data = data.to(device)
         net.to(device)
         out = net(data)
+        # for each spectrogram/row index of max probability
+        pred = np.argmax(out.detach().cpu().numpy(), axis=1)
+        # find most frequent index over all spectrograms
+        consensus = np.bincount(pred).argmax()
```
Comment (birdNicolas) [inline predict.py:92]: Versuche die Implementierung nachzuvollziehen...
Es wird für jeden Zeit-Frame eine unabhängige Vorhersage gemacht.
` out = net(data)`
Und ich sehe, da gerade 2 Implementierungen, um die finale Vogelklasse zu bestimmen:
(1) für jeden Zeit-Frame bestimmte ich das Maximum und nehme die Klasse mit denen meisten Votes
 ```
pred = np.argmax(out.detach().cpu().numpy(), axis=1)
consensus = np.bincount(pred).argmax()
```
(2) man nehme das Mean über alle Predictions über die Zeitachse und die `argmax` davon ist die Vogelklasse:
``` 
mean_probs = np.mean(out.detach().cpu().numpy(), axis=0)
idx = np.argmax(mean_probs)
```
Richtig soweit?
Momentan wird Variante (1) verwendet. Was spricht dafür? Gibt es eine bessere Accuracy?

File: predict.py [hunk 3]
```diff
@@ -78,21 +78,27 @@ def predict(net, labels, files, params):
     predictions = []
     probs = []
     for i, file in enumerate(files):
-        processed = file.split('.')[0] + '_proc.wav'
+        filename = os.path.splitext(os.path.basename(file))[0]
+        processed = filename + '_proc.wav'
         pre.preprocess(file, processed)
         data = vggish_input.wavfile_to_examples(processed)
         data = torch.from_numpy(data).unsqueeze(1).float()
```
Comment (birdNicolas) [inline predict.py:85]: Bei mir hatte `data` die Dimensionen `28, 1, 96, 64`.
Zum Verständnis:
- 28 sind der Anzahl der Zeit-Bins
- 96 werden die Spectogramgröße sein
Richtig?

Fragen:
- Warum ist 28 an erster Stelle? Hätte gedacht, die erste Stelle sollte die Batch-Axis sein?
- Wofür steht die 64?

File: predict.py (hunk 2 above)
Comment (lucasmllr) [reply predict.py:92]: das mit dem consensus war quatsch. das kann komplett raus. weiß nicht mehr was uns dazu getrieben hat. mitteln der wahrscheinlichkeiten über alle Zeitfenster und dann max stimmt ja auf jeden fall

File: predict.py (hunk 3 above)
Comment (lucasmllr) [reply predict.py:85]: 28 ist die Anzahl der Zeitfenster und kommt hier an die Stelle der Batchsize. Spricht da was gegen? in der predict methode werden dann die wahrscheinlichkeiten über diese axe gemittelt. 96 ist die Anzahl der Zeitbins der einzelnen Spektrogramme, 64 die Anzahl der Frequenzbins.

Comment (henriettemiko) [reply predict.py:85]: Ich hatte das so verstanden, dass das input wav file in (in deinem Fall) 28 Spektrogramme  (96x64) geteilt wird. Müsste aber auf die Länge des Files ankommen, in wieviele, oder?  Für jedes der Spektrogramme werden die Wahrscheinlichkeiten der Klassen berechnet, also 28 x 10. Dann mean der Wahrscheinlichkeiten (1x10) und davon das max als prediction.

File: predict.py (hunk 2 above)
Comment (henriettemiko) [reply predict.py:92]: ja, der code zu (1) kann dann raus
